data-collection/src/main/java/refactoringml/ProcessMetricsCollector.java

```diff
@@ -51,73 +47,44 @@ public ProcessMetricsCollector(Project project, Database db, Repository reposito
 		this.branch = branch;
 		this.fileStoragePath = FilePathUtils.lastSlashDir(fileStoragePath);
 		this.lastCommitToProcess = lastCommitToProcess;
-		todo = new HashMap<>();
 		pmDatabase = new PMDatabase(commitThreshold);
 	}
 
-	public void addToList (RevCommit commitData, Yes yes) {
-		String id = commitData.getName();
-		if(!todo.containsKey(id))
-			todo.put(id, new HashSet<>());
-
-		todo.get(id).add(yes.getId());
-	}
-
-	public void collect() throws IOException {
-
-		RevWalk walk = JGitUtils.getReverseWalk(repository, branch);
-
-		RevCommit commit = walk.next();
-
-		boolean lastFound = false;
-		while(commit!=null && !lastFound) {
-
-			// did we find the commit to stop?
-			// if so, process it, and then stop
-			if(commit.getName().equals(lastCommitToProcess))
-				lastFound = true;
-
-			if(commit.getParentCount() <= 1) {
-
-				log.debug("Commit ID " + commit.getName());
-				RevCommit commitParent = commit.getParentCount() == 0 ? null : commit.getParent(0);
+	public void collectMetrics(RevCommit commit, Set<Long> allYeses, boolean isRefactoring) throws IOException {
+		if(commit.getParentCount() > 1) return;
 
-				Set<String> refactoredClasses = new HashSet<>();
-
-				// if the class happened to be refactored, then, print its process metrics at that time
-				if (todo.containsKey(commit.getName())) {
-					try {
-						db.openSession();
-						refactoredClasses = collectProcessMetricsOfRefactoredCommit(commit);
-						db.commit();
-					} catch (Exception e) {
-						log.error("Error when collecting process metrics in commit " + commit.getName(), e);
-						db.rollback();
-					} finally {
-						db.close();
-					}
-				}
+		RevCommit commitParent = commit.getParentCount() == 0 ? null : commit.getParent(0);
+		Set<String> refactoredClasses = new HashSet<>();
 
-				// we go now change by change in the commit to update the process metrics there
-				// (no need for db here, as this update happens only locally)
-				updateProcessMetrics(commit, commitParent);
-
-				// update classes that were not refactored on this commit
-				try {
-					db.openSession();
-					updateAndPrintExamplesOfNonRefactoredClasses(commit, refactoredClasses);
-					db.commit();
-				} catch (Exception e) {
-					db.rollback();
-				} finally {
-					db.close();
-				}
+		// if the class happened to be refactored, then, print its process metrics at that time
```

Comment:
maybe the comment here should be about "if this commit contained a refactoring, then collect its process metrics"